src/analysis/processing/qgsrasteranalysisutils.h

```diff
@@ -65,6 +69,21 @@ namespace QgsRasterAnalysisUtils
    * \see createRasterTypeParameter()
    */
   Qgis::DataType rasterTypeChoiceToDataType( int choice );
+
+  struct BooleanLogicInput
+  {
+    std::unique_ptr< QgsRasterInterface > sourceDataProvider;
+    std::unique_ptr< QgsRasterProjector> projector;
+    QgsRasterInterface *interface = nullptr;
+    bool hasNoDataValue = false;
+    int band = 1;
```

Comment:
Could this be a QList< int > parameter instead, to make it more flexible?

Reply:
yep - but it's not part of stable api, so we can change this anytime we want/need :)